src/components/Table/index.tsx

```diff
@@ -0,0 +1,302 @@
+import * as React from 'react';
+import { TableHead } from './TableHead';
+import { TableToolbar } from './TableToolbar';
+import type { AutoUIEntityPropertyDefinition } from '../../AutoUI';
+import type { CheckedState, Pagination, TableSortOptions } from './utils';
+import { DEFAULT_ITEMS_PER_PAGE } from '../../AutoUI/utils';
+import { Material } from '@balena/ui-shared-components';
+import { Divider } from 'rendition';
+
+const {
+	Box,
+	Table: MaterialTable,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TablePagination,
+	TableRow,
+	Paper,
+	Checkbox,
+	styled,
+} = Material;
+
+const StyledMaterialTable = styled(MaterialTable)(() => ({
+	'& [data-table="table_cel__sticky"]': {
+		position: 'sticky',
+		left: 0,
+		zIndex: 9,
+		backgroundColor: 'inherit',
+	},
+	'& [data-table="table_cel__sticky_header"]': {
+		position: 'sticky',
+		left: 0,
+		zIndex: 10,
+	},
+}));
+
+interface TableProps<T> {
+	rowKey: keyof T;
+	data: T[];
+	checkedItems?: T[];
+	checkedState?: CheckedState;
+	columns: Array<AutoUIEntityPropertyDefinition<T>>;
+	pagination: Pagination;
+	sort: TableSortOptions;
+	actions?: Material.MenuItemProps[];
+	onCheck?: (selected: T[] | undefined, allChecked?: CheckedState) => void;
+	onSort?: (sort: TableSortOptions) => void;
+	onRowClick?: (
+		entity: T,
+		event: React.MouseEvent<HTMLTableCellElement, MouseEvent>,
+	) => void;
+	onPageChange?: (page: number, itemsPerPage: number) => void;
+	onColumnPreferencesChange?: (
+		columns: Array<AutoUIEntityPropertyDefinition<T>>,
+	) => void;
+}
+
+export const Table = <T extends object>({
+	rowKey,
+	data,
+	checkedItems = [],
+	checkedState,
+	columns,
+	pagination,
+	sort,
+	actions,
+	onCheck,
+	onSort,
+	onRowClick,
+	onPageChange,
+	onColumnPreferencesChange,
+}: TableProps<T>) => {
+	const lastSelected = React.useRef<T>();
+
+	const totalItems = pagination?.totalItems || data.length;
```

Comment:
not `??`? In this case if `totalItems` is `0` it tries to use `data.length` instead, but I would expect that if we pass a value for `totalItems` it should be obeyed over anything else